### PR TITLE
doc(transports): Improvements

### PR DESCRIPTION
**Pull request by Guikingone**

### CODE REVIEW

File: doc/transport.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # Transport
 
-This bundle provides multiple transports to store and handle tasks.
+Transports are the foundations behind the storage of tasks, once tasks are defined in the configuration
+or scheduled via the [Scheduler](../src/Scheduler.php), every task is stored via transports.
+
+Think of transport like the transports used by the following components in Symfony
```
Comment (jmsche) [inline doc/transport.md:6]:
```suggestion
Think of transports like the transports used by the following components in Symfony:
```

File: doc/transport.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+
+- Mailer
+- Messenger
+- Notifier
+
+This bundle defines a set of transports, each transport has its own configuration and can be override if required.
```
Comment (jmsche) [inline doc/transport.md:12]:
```suggestion
This bundle defines a set of transports, each transport has its own configuration and can be overridden if required. 
```

File: doc/transport.md [hunk 3]
```diff
@@ -66,6 +94,8 @@ scheduler_bundle:
         dsn: 'failover://(memory://first_in_first_out || memory://last_in_first_out)' # Or 'fo://(memory://first_in_first_out || memory://last_in_first_out)'
 ```
 
+**configuration**: This transport requires at least 2 transport to be used (each one can be configured as normal).
```
Comment (jmsche) [inline doc/transport.md:97]:
```suggestion
**configuration**: This transport requires at least 2 transports to be used (each one can be configured as normal).
```

File: doc/transport.md [hunk 4]
```diff
@@ -79,6 +109,8 @@ scheduler_bundle:
         dsn: 'longtail://(memory://first_in_first_out <> memory://last_in_first_out)' # Or 'lt://(memory://first_in_first_out <> memory://last_in_first_out)'
 ```
 
+**configuration**: This transport requires at least 2 transport to be used (each one can be configured as normal).
```
Comment (jmsche) [inline doc/transport.md:112]:
```suggestion
**configuration**: This transport requires at least 2 transports to be used (each one can be configured as normal).
```

File: doc/transport.md [hunk 5]
```diff
@@ -93,6 +125,8 @@ scheduler_bundle:
         dsn: 'roundrobin://(memory://first_in_first_out && memory://last_in_first_out)' # Or 'rr://(memory://first_in_first_out && memory://last_in_first_out)'
 ```
 
+**configuration**: This transport requires at least 2 transport to be used (each one can be configured as normal).
```
Comment (jmsche) [inline doc/transport.md:128]:
```suggestion
**configuration**: This transport requires at least 2 transports to be used (each one can be configured as normal).
```

**Review by jmsche — COMMENTED** [review]
> each one can be configured as normal

"as usual"?

File: doc/transport.md (hunk 1 above)
Comment (Guikingone) [reply doc/transport.md]: Fixed

File: doc/transport.md (hunk 3 above)
Comment (Guikingone) [reply doc/transport.md]: Fixed

File: doc/transport.md (hunk 5 above)
Comment (Guikingone) [reply doc/transport.md]: Fixed

File: doc/transport.md (hunk 4 above)
Comment (Guikingone) [reply doc/transport.md]: Fixed

File: doc/transport.md (hunk 2 above)
Comment (Guikingone) [reply doc/transport.md]: Fixed

**Comment by Guikingone** [comment]
> > each one can be configured as normal
> 
> "as usual"?

Nice catch, fixed 🙂

File: doc/worker.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- As a daemon (using the `--wait` option in the consume command)
+- As a loop (default behaviour)
+
+## Daemon
+
+The worker has been built to be able to "wait" due tasks, the approach is simple:
+
+- Once launched, the worker ask the scheduler to returns due tasks
+- If there's no due tasks, the worker use the current time and determine the "wait" period until the next minute.
+- If there's due tasks, the worker will call the registered runners to execute each task, once every task
```
Comment (jmsche) [inline doc/worker.md:19]: "If there's" => "If there are" (twice)

**Review by jmsche — CHANGES_REQUESTED** [review]

File: doc/worker.md (hunk 6 above)
Comment (Guikingone) [reply doc/worker.md]: Nice catch, fixed

File: doc/worker.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+- As a daemon (using the `--wait` option in the consume command)
+- As a loop (default behaviour)
+
+## Daemon
+
+The worker has been built to be able to "wait" due tasks, the approach is simple:
+
+- Once launched, the worker ask the scheduler to returns due tasks
+- If there's no due tasks, the worker use the current time and determine the "wait" period until the next minute.
```
Comment (jmsche) [inline doc/worker.md:18]:
```suggestion
- If there are no due tasks, the worker uses the current time and determines the "wait" period until the next minute.
```

File: doc/worker.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+The worker uses a very simple approach and can act in two ways:
+
+- As a daemon (using the `--wait` option in the consume command)
+- As a loop (default behaviour)
+
+## Daemon
+
+The worker has been built to be able to "wait" due tasks, the approach is simple:
+
+- Once launched, the worker ask the scheduler to returns due tasks
```
Comment (jmsche) [inline doc/worker.md:17]:
```suggestion
- Once launched, the worker asks the scheduler to return due tasks
```

File: doc/worker.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+- If there's no due tasks, the worker use the current time and determine the "wait" period until the next minute.
+- If there are due tasks, the worker will call the registered runners to execute each task, once every task
```
Comment (jmsche) [inline doc/worker.md:19]:
```suggestion
- If there are due tasks, the worker will call the registered runners to execute each task. Once every task
```

File: doc/transport.md [hunk 10]
```diff
@@ -1,6 +1,41 @@
 # Transport
 
-This bundle provides multiple transports to store and handle tasks.
+Transports are the foundations behind the storage of tasks, once tasks are defined in the configuration
```
Comment (jmsche) [inline doc/transport.md:3]:
```suggestion
Transports are the foundations behind the storage of tasks: once tasks are defined in the configuration
```

File: doc/transport.md [hunk 11]
```diff
@@ -1,6 +1,41 @@
...
+final class FooController
+{
+    public function __invoke(Request $request, TransportInterface $transport)
+    {
+        // ...
+    }
+}
+```
+
+_Note: Using the transport without the scheduler can lead to edge issues as the scheduler is synchronized with it_
```
Comment (jmsche) [inline doc/transport.md:38]:
```suggestion
_Note: Using the transport without the scheduler can lead to edge issues as the scheduler is synchronized with it._
```

File: doc/transport.md [hunk 12]
```diff
@@ -35,11 +73,25 @@ scheduler_bundle:
...
+  
+- The second and optional key is the `path` where every task is stored (using `json` format):
+
+```yaml
+scheduler_bundle:
+    transport:
+        dsn: 'filesystem://first_in_first_out?path=/srv/app'
+```
+
+_Note: Container parameters cannot be passed here as the container is not involved in the transport configuration_
```
Comment (jmsche) [inline doc/transport.md:89]:
```suggestion
_Note: Container parameters cannot be passed here as the container is not involved in the transport configuration._
```

File: doc/transport.md [hunk 13]
```diff
@@ -66,11 +120,14 @@ scheduler_bundle:
         dsn: 'failover://(memory://first_in_first_out || memory://last_in_first_out)' # Or 'fo://(memory://first_in_first_out || memory://last_in_first_out)'
 ```
 
+**configuration**: This transport requires at least 2 transports to be used (each one can be configured as usual).
+
 ## LongTail
 
-The [LongTail](../src/Transport/LongTailTransport.php) allows to use multiple transport and it specifically designed
+The [LongTail](../src/Transport/LongTailTransport.php) allows to use multiple transport, it's specifically designed
```
Comment (jmsche) [inline doc/transport.md:127]:
```suggestion
The [LongTail](../src/Transport/LongTailTransport.php) allows to use multiple transports. It's specifically designed
```

File: doc/transport.md [hunk 14]
```diff
@@ -66,11 +120,14 @@ scheduler_bundle:
...
 to maximize the transport usage by always trying to use the transport with the lowest amount of tasks, 
 this approach can help when you're scheduling tasks in a [high-stress environment](https://en.wikipedia.org/wiki/Long_tail).
```
Comment (jmsche) [inline doc/transport.md:129]:
```suggestion
to maximize the transport usage by always trying to use the transport with the lowest amount of tasks. 
This approach can help when you're scheduling tasks in a [high-stress environment](https://en.wikipedia.org/wiki/Long_tail).
```

File: doc/transport.md [hunk 15]
```diff
@@ -93,19 +152,30 @@ scheduler_bundle:
...
+**configuration**: This transport requires at least 2 transports to be used (each one can be configured as usual).
+
 ## Redis
 
-The [RedisTransport](../src/Bridge/Redis/Transport/RedisTransport.php) allows to use Redis as tasks storage.
-This transport is useful if you need to share tasks between multiple projects/instances.
+The [RedisTransport](../src/Bridge/Redis/Transport/RedisTransport.php) allows to use Redis 
+as tasks storage, this transport is useful if you need to share tasks between multiple projects/instances.
+
+**Requirements**: The `Redis` transport requires redis >= 4.3 at least.
```
Comment (jmsche) [inline doc/transport.md:162]:
```suggestion
**Requirements**: The `Redis` transport requires at least redis >= 4.3.
```

File: doc/transport.md [hunk 16]
```diff
@@ -93,19 +152,30 @@ scheduler_bundle:
...
 scheduler_bundle:
     transport:
-        dsn: 'redis://first_in_first_out'
+        dsn: 'redis://user:password@host:port'
 ```
 
+**configuration**: This transport requires multiple options:
+
+- The `user` of the Redis instance
+- The `password` related
```
Comment (jmsche) [inline doc/transport.md:175]:
```suggestion
- The related `password`
```

File: doc/transport.md [hunk 17]
```diff
@@ -93,19 +152,30 @@ scheduler_bundle:
...
+- The `host` of the Redis instance
+- The optional `port` of the Redis instance
+
 ### Extra configuration
 
 This transport can configure multiple options:
```
Comment (jmsche) [inline doc/transport.md:181]:
```suggestion
This transport can be configured using multiple options:
```

**Review by jmsche — CHANGES_REQUESTED** [review]

**Comment by Guikingone** [comment]
@jmsche Are we good on this one?

**Comment by jmsche** [comment]
I think so :)

**Review by jmsche — APPROVED** [review]

**Comment by Guikingone** [comment]
Merged, thanks @jmsche 🎉

**Pull request merged**